pkg/controller/storagecluster/kms_resources.go

```diff
@@ -0,0 +1,56 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *ReconcileStorageCluster) ensureKMSSystem(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// if there is no 'Security' spec to inspect, return silently
+	if instance.Spec.Security == nil {
+		return nil
+	}
+	if instance.Spec.Security.KeyManagementService != nil &&
+		instance.Spec.Security.KeyManagementService.ConnectionDetails == "" {
+		// if there is 'ConnectionDetails' configmap name is not provided, then return silently
+		return nil
+	}
+	ctx := context.TODO()
+	// Check for the CephCluster
+	found := &cephv1.CephCluster{}
+	err := r.client.Get(ctx,
+		types.NamespacedName{
+			Name:      generateNameForCephCluster(instance),
+			Namespace: instance.Namespace,
+		},
+		found,
+	)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("CephCluster yet to start."))
+		return err
+	}
+	kmsConfigMapName := strings.TrimSpace(instance.Spec.Security.KeyManagementService.ConnectionDetails)
```

Comment:
Let check the CM before fetching the `CephCluster` so if it fails, it's one less API call... (no need to call for `CephCluster`)

Reply:
Changed this part, please take a look.